pkg/node/node.go

```diff
@@ -350,11 +353,19 @@ func NewBee(o Options) (*Bee, error) {
 					return
 				}
 
-				err = addressbook.Put(overlay, addr)
+				bzzAddr, err := bzz.NewAddress(signer, addr, overlay, o.NetworkID)
+				if err != nil {
+					_ = p2ps.Disconnect(overlay)
+					logger.Debugf("new bzz address error %s %s: %v", a, overlay, err)
+					logger.Errorf("connect to bootnode %s", a)
+					return
+				}
+
+				err = addressbook.Put(overlay, *bzzAddr)
 				if err != nil {
 					_ = p2ps.Disconnect(overlay)
 					logger.Debugf("addressbook error persisting %s %s: %v", a, overlay, err)
-					logger.Errorf("persisting node %s", a)
+					logger.Errorf("connect to bootnode %s", a)
```

Comment:
I feel that to the user, it's in the scope of failed connect, as it will result in failed connect attempt. If the error message is `persisting node`, I believe it is not really user friendly, so I left it in Debug, for debugging. But I can also switch back, let me know what you think.